### PR TITLE
nautilus-open-in-blackbox: init at 0.1.1

### PR DESCRIPTION
**Pull request by BlankParticle**
## Description of changes
A Nautilus Plugin to add Open in Blackbox in Context Menu

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/na/nautilus-open-in-blackbox/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  pname = "nautilus-open-in-blackbox";
+  version = "0.1.1";
+
+  src = fetchFromGitHub
+    {
+      owner = "ppvan";
+      repo = pname;
+      rev = version;
+      hash = "sha256-5rvh3qNalpjamcBVQrnAW6GxhwPPlRxP5h045YDqvrM=";
+    };
```
Comment (adamcstephens) [inline pkgs/by-name/na/nautilus-open-in-blackbox/package.nix:12]:
```suggestion
  src = fetchFromGitHub {
    owner = "ppvan";
    repo = pname;
    rev = "refs/tags/${version}";
    hash = "sha256-5rvh3qNalpjamcBVQrnAW6GxhwPPlRxP5h045YDqvrM=";
  };
```

File: pkgs/by-name/na/nautilus-open-in-blackbox/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  patches = [ ./paths.patch ];
+
+  buildInputs = [
+    gnome.nautilus-python
+    python3.pkgs.pygobject3
+  ];
+
+  installPhase = ''
```
Comment (eclairevoyant) [inline pkgs/by-name/na/nautilus-open-in-blackbox/package.nix:21]:
```suggestion
  installPhase = ''
    runHook preInstall

```
always preserve pre/post phase hooks when overriding a phase

File: pkgs/by-name/na/nautilus-open-in-blackbox/package.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    mkdir -p $out/share/nautilus-python/extensions
+    cp ./nautilus-open-in-blackbox.py $out/share/nautilus-python/extensions
```
Comment (eclairevoyant) [inline pkgs/by-name/na/nautilus-open-in-blackbox/package.nix:23]:
```suggestion
    install -Dm555 ./nautilus-open-in-blackbox.py -t $out/share/nautilus-python/extensions
    
    runHook postInstall
```
and delete the `mkdir` line

File: pkgs/by-name/na/nautilus-open-in-blackbox/package.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ python3, fetchFromGitHub, gnome, stdenv, lib }:
+stdenv.mkDerivation rec {
+  pname = "nautilus-open-in-blackbox";
+  version = "0.1.1";
+
+  src = fetchFromGitHub
+    {
+      owner = "ppvan";
+      repo = pname;
```
Comment (eclairevoyant) [inline pkgs/by-name/na/nautilus-open-in-blackbox/package.nix:9]:
```suggestion
      repo = "nautilus-open-in-blackbox";
```
as repo and pname are semantically different

File: pkgs/by-name/na/nautilus-open-in-blackbox/package.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      rev = version;
+      hash = "sha256-5rvh3qNalpjamcBVQrnAW6GxhwPPlRxP5h045YDqvrM=";
+    };
+
+  patches = [ ./paths.patch ];
```
Comment (eclairevoyant) [inline pkgs/by-name/na/nautilus-open-in-blackbox/package.nix:14]: patches always need comments, and fetching from upstream is preferable to vendoring, when possible

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/by-name/na/nautilus-open-in-blackbox/package.nix (hunk 1 above)
Comment (adamcstephens) [reply pkgs/by-name/na/nautilus-open-in-blackbox/package.nix]: Part of this suggestion was moving the opening brace in-line with fetchFromGitHub and adjusting indentation accordingly.. While we don't have official formatting yet, and I'm not a fan of nit-picking formatting, I do think it's important to understand this is the de facto standard in nixpkgs.

Comment (BlankParticle) [reply pkgs/by-name/na/nautilus-open-in-blackbox/package.nix]: Sorry, I looked over the formatting, I think It was auto formatted. Anyways I have fixed the formatting

**Review by adamcstephens — APPROVED** [review]

**Comment by linsui** [comment]
FYI, there is already a nautilus-open-any-terminal which works for blackbox and actively maintained.

**Comment by BlankParticle** [comment]
> FYI, there is already a nautilus-open-any-terminal which works for blackbox and actively maintained.

Yeah I know it exists, but I couldn't get that to work. So I ported this instead

**Comment by linsui** [comment]
Maybe we should write a module for that.

**Comment by linsui** [comment]
Here you are. https://github.com/NixOS/nixpkgs/pull/283778 :)

**Pull request merged**